### PR TITLE
N+1 integration test

### PR DESCRIPTION
**Pull request by shaaibu7**
- [ ] Use Capybara to write integration tests for each view.
- [ ] User index page:

      I can see the username of all other users.
      I can see the profile picture for each user.
      I can see the number of posts each user has written.
      When I click on a user, I am redirected to that user's show page.
      

- [ ] user show page:

      I can see the user's profile picture.
      I can see the user's username.
      I can see the number of posts the user has written.
      I can see the user's bio.
      I can see the user's first 3 posts.
      I can see a button that lets me view all of a user's posts.
      When I click a user's post, it redirects me to that post's show page.
      When I click to see all posts, it redirects me to the user's post's index page.

- [ ] User post index page:

      I can see the user's profile picture.
      I can see the user's username.
      I can see the number of posts the user has written.
      I can see a post's title.
      I can see some of the post's body.
      I can see the first comments on a post.
      I can see how many comments a post has.
      I can see how many likes a post has.
      I can see a section for pagination if there are more posts than fit on the view.
      When I click on a post, it redirects me to that post's show page.

- [ ] Post show page:

      I can see the post's title.
      I can see who wrote the post.
      I can see how many comments it has.
      I can see how many likes it has.
      I can see the post body.
      I can see the username of each commentor.
      I can see the comment each commentor left.

Screenshot from 2023-05-30 10-38-14
Screenshot from 2023-05-30 10-41-08

### CODE REVIEW

File: spec/integration/post_spec.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 RSpec.describe 'Posts', type: :feature do
   before(:each) do
-    @user = User.create(name: 'John', photo: 'www.example.com', bio: 'Life science', posts_counter: 0)
+    @user = User.create(name: 'John', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Life science',
+                        posts_counter: 0)
```
Comment (GzimAsani) [inline spec/integration/post_spec.rb:6]: - [ ] You have 7 failing tests, kindly check into this 
image

Comment (GzimAsani) [inline spec/integration/post_spec.rb:6]: - [ ] Kindly work on your controllers in order to fix the n+1 issue and also In your PR description include the screenshots of your app's logs before and after the fix 
image

Comment (GzimAsani) [inline spec/integration/post_spec.rb:6]: - [ ] The tests files need to be separated for each page, kindly update this
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, TEAM            :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with the tests, you've added all the test cases that were required :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    # @postthree = Post.create(title: 'good', text: 'journey', author: @user, comments_counter: 0, likes_counter: 0)
+    # @postfour = Post.create(title: 'amazing', text: 'space travel', author: @user, comments_counter: 0,
+    #                         likes_counter: 0)
+  end
+
+  describe 'Test for index page of users' do
+    it 'should test for correct content on the page' do
+      visit '/'
+      expect(page).to have_content(@user.name)
+    end
```
Comment (rloterh) [inline spec/integration/user_index_spec.rb:18]: - [ ] [OPTIONAL]
A test to assert that the scenario **_I can see the username of all other users_**. for _user_ _index_ spec is missing! Please ensure to improve upon this aspect.

File: spec/integration/post_spec.rb (hunk 1 above)
Comment (rloterh) [reply spec/integration/post_spec.rb:6]: well resolved! ✔️

Comment (rloterh) [reply spec/integration/post_spec.rb:6]: well resolved! ✔️

Comment (rloterh) [reply spec/integration/post_spec.rb:6]: Needs attention!

File: spec/integration/user_index_spec.rb (hunk 2 above)
Comment (rloterh) [inline spec/integration/user_index_spec.rb]: - [ ] [OPTIONAL]
A test to assert that the scenario **_I can see the username of all other users_**. for _user_ _index_ spec is missing! Please ensure to improve upon this aspect.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Suleiman, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!
🏁 🚦 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Nice job improving on the code!
- ReadMe looks great!

## Required Changes ♻️
- [ ] Nice job for showing solutions to `N + 1` solved from the console. However, we actually want you to apply the solution in your actual code (_controllers_/_model_), for instance, modifying a fetch query in the _post_ _index_ actions, similar to:

```ruby
   @posts = @user.posts.includes(:comments)
```

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @shaaibu7 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All tests passed 👍 
* N+1 problems solved 👍 
* No linter errors 👍 
* Integration tests written with capybara gem 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**